### DISCUSSION TITLE
no runtime found in opensuse world

### DISCUSSION THREAD

**Discussion by uciharis** [helix-editor/helix#10541] [Troubleshooting] [answered]
### Summary

im using opensuse Leap 15.5 and installed helix from 2 sources :
- from Yast(zypper) package manager
- from brew
every time i ran hx --health or helix --health

the message notified that runtime directory doesnt exist

### Reproduction Steps

 

I tried this:

1. `hx --health`

~> Log file: /home/el490/.cache/helix/helix.log
Runtime directories: /home/el490/.config/helix/runtime;/home/linuxbrew/.linuxbrew/Cellar/helix/24.03/libexec/runtime;/home/linuxbrew/.linuxbrew/Cellar/helix/24.03/bin/runtime
Runtime directory does not exist: /home/el490/.config/helix/runtime
Runtime directory does not exist: /home/linuxbrew/.linuxbrew/Cellar/helix/24.03/bin/runtime
Clipboard provider: xclip
System clipboard provider: xclip

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
0 bytes and no text i found here
```

</details>

### Platform

opensuse Leap 15.5

### Terminal Emulator

gnome terminal version 3.42.3 gnome 41

### Installation Method

package manager yast zypper and brew

### Helix Version

helix 24.3(2cadec0b)

**Comment by pascalkuthe** [comment, marked as answer]
Sounds like an issue with the OpenSuse package. You need to report that against their issue tracker. We don't have control over those

**Reply by uciharis** [reply]
thankyou. 
i also found this problem in fedora 40, so maybe the packager forgot something to add

**Comment by LJolmes** [comment]
I figured out what the problem is and I can provide a hotfix, even though I'm not sure about all of the consequences.
Basically, when you install the runtime package, its making a runtime folder in /usr/lib64/helix/runtime and adds a symlink in ~/.config/helix/runtime. The result is, that the path for the runtime files from the package is now ~/config/helix/runtime/runtime/ and therefore helix won't find them. This might be due to the fact that there's already a runtime folder in the .config/helix directory with some content (grammar folder) at the time the symlink is generated. In that case, the symlink is added into the folder rather than deleting the content. However, the way to fix it is to get the contents from /usr/lib64/helix/runtime to ~/.config/helix. What you could do is, in the ~/.config/helix directory:
`mv runtime runtime-old`
`cp runtime-old/runtime runtime`
That way you fix the link, and helix is only using the runtime files provided by the package. If you want to modify them, you'd have to use sudo and any changes will most likely get lost with the next package update.
Another alternative would be symlinking the contents from /usr/lib64/helix/runtime that you need into ~/.config/helix/runtime instead of using a single symlink to the base runtime package directory. That way you could use e.g. queries and themes from the runtime package but stick to your own grammar folder in case you modified something there.

**Reply by uciharis** [reply]
thank u Ljolmes. i have checked the runtime targeted folder based on your reply but i dont get it in my directory. i think the packager from the distro miss that think when shipping this editor to their repository

**Comment by rim99** [comment]
Another workaround is to set env in the `~/.profile`, `~/.bashrc` or `~/.zshrc`

```
export HELIX_RUNTIME="/usr/lib64/helix/runtime"
``` 

It would allow helix to use installed dir

**Comment by rim99** [comment]
Another workaround is to set env in the `~/.profile`, `~/.bashrc` or `~/.zshrc`

```
export HELIX_RUNTIME="/usr/lib64/helix/runtime"
``` 

It would allow helix to use installed dir
